nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,129 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.rest.management.elasticsearch;
+
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.elasticsearch.ElasticsearchIndexOperation;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.model.Repository;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "elasticsearch")
+public class ElasticsearchObject extends DefaultObject {
+
+    /**
+     * Runs an ES indexation on the given repository.
+     *
+     * @param repositoryName the repository name for which we want to start the ES indexation
+     * @throws NuxeoException
+     */
+    @POST
+    @Path("{repositoryName}/reindex")
+    public void doIndexing(@PathParam("repositoryName") String repositoryName) {
+        Repository repository = getRepository(repositoryName);
+        Framework.doPrivileged(() -> {
+            try (CloseableCoreSession coreSession = CoreInstance.openCoreSessionSystem(repository.getName())) {
+                AutomationService automationService = Framework.getService(AutomationService.class);
+                automationService.run(new OperationContext(coreSession), ElasticsearchIndexOperation.ID);
+            } catch (OperationException oe) {
+                throw new NuxeoException(oe);
+            }
+        });
+    }
+
+    /**
+     * Executes an ES flush on all documents indexes of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/flush")
+    public void doFlush(@PathParam("repositoryName") String repositoryName) {
+        Repository repository = getRepository(repositoryName);
+        Framework.doPrivileged(() -> {
+            Framework.getService(ElasticSearchAdmin.class).flushRepositoryIndex(repository.getName());
+        });
+    }
+
+    /**
+     * Executes an ES optimize on all documents indexes of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/optimize")
+    public void doOptimize(@PathParam("repositoryName") String repositoryName) {
+        Repository repository = getRepository(repositoryName);
+        Framework.doPrivileged(() -> {
```

Comment:
No need to use curly bracket for lambda, right?

Reply:
sure, thx, a bad copy/past